### ISSUE TITLE
Grayscale code highlighter

### ISSUE THREAD

**Issue by mitjafelicijan** [asciidoctor/asciidoctor#2494]
I am using asciidoctor to write a book and am in need of grayscale code highlighter. I am currently using rouge and tried coderay but none of them is grayscale.

Is there any code highlighter that outputs grayscale code such as picture below?

![screenshot from 2017-11-21 18-16-17](https://user-images.githubusercontent.com/296714/33086617-60431e8e-cee8-11e7-9e60-860d1fd284b1.png)

**Comment by mojavelinux** [comment]
Pygments has a grayscale theme named "bw". We use it in Asciidoctor EPUB3.

Rouge doesn't yet have a bw theme, but it's pretty straightforward to port a theme from Pygments to Rouge. Therefore, you could consider contributing it to Rouge.

Here's how that might look:

bw.rb
```
require 'rouge' unless defined? ::Rouge.version

module Rouge; module Themes
  class BlackWhiteTheme < CSSTheme
    name 'bw'

    style Text,                        :fg => '#000000', :bg => '#ffffff'

    style Comment,                     :italic => true
    style Comment::Preproc,            :italic => false

    style Keyword,                     :bold => true
    style Keyword::Pseudo,             :bold => false
    style Keyword::Type,               :bold => false

    style Operator,                    :bold => true

    style Name::Class,                 :bold => true
    style Name::Namespace,             :bold => true
    style Name::Exception,             :bold => true
    style Name::Entity,                :bold => true
    style Name::Tag,                   :bold => true

    style Literal::String,             :italic => true
    style Literal::String::Interpol,   :bold => true
    style Literal::String::Escape,     :bold => true

    style Generic::Heading,            :bold => true
    style Generic::Subheading,         :bold => true
    style Generic::Emph,               :italic => true
    style Generic::Strong,             :bold => true
    style Generic::Prompt,             :bold => true

    style Error,                       :fg => '#ffffff', :bg => '#000000'
  end
end; end
```

**Comment by mojavelinux** [comment]
Now that Rouge integration is merged into Asciidoctor, this will be achievable starting in Asciidoctor 2.0. Here are the steps:

1. Download the bw.rb file from https://github.com/jneen/rouge/blob/861f06ebef3636a8db606d1f8c6d85d12199b2f8/lib/rouge/themes/bw.rb
2. Invoke Asciidoctor as follows:
   ```
   asciidoctor -r ./bw.rb -a source-highlighter=rouge -a rouge-style=bw document.adoc
   ```

Once the bw theme is merged into Rouge, step 1 won't be necessary.
